packages/artifact_proxy/lib/src/artifact_manifest_client.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:artifact_proxy/artifact_proxy.dart';
+import 'package:checked_yaml/checked_yaml.dart';
+import 'package:http/http.dart' as http;
+
+/// {@template artifact_manifest_client}
+/// A client that fetches [ArtifactsManifest]s from the shorebird storage bucket
+/// and caches them in memory.
+/// {@endtemplate}
+class ArtifactManifestClient {
+  /// {@macro artifact_manifest_client}
+  ArtifactManifestClient({http.Client? httpClient})
+      : _httpClient = httpClient ?? http.Client();
+
+  final http.Client _httpClient;
+
+  final _cache = <String, ArtifactsManifest>{};
```

Comment:
You'll want this to be a fixed-size LRU at some point.

Reply:
done 👍